src/dataflow/src/sink/kafka.rs

```diff
@@ -610,9 +611,19 @@ where
                 name.clone(),
             )
         }
-        _ => {
-            // this will not be true once we add new kafka sink formats!
-            unreachable!("kafka sink without schemas")
+        None => {
```

Comment:
nit: it's weird that `Some(PublishedSchemaInfo)` means Avro, and `None` means JSON. But it sounded like you would address that next, so feel free to ignore.